docs/designs/kusto-to-metrics.md

```diff
@@ -6,91 +6,179 @@ ADX-Mon currently provides two distinct capabilities:
 2. **OTLP Metrics Exporters** - Export metrics to external observability systems ([`collector/export/metric_otlp.go`](https://github.com/Azure/adx-mon/blob/main/collector/export/metric_otlp.go))
 
-However, there's no mechanism to connect these two capabilities. Organizations often need to:
-- Transform aggregated ADX data (computed via SummaryRules) into standardized OTLP metrics
+However, there's no direct mechanism to execute KQL queries and export the results as metrics to observability platforms. Organizations often need to:
+- Execute KQL queries on ADX data and export results as standardized metrics
 - Export these metrics to external systems (Prometheus, DataDog, etc.) at regular intervals
 - Create derived metrics from complex KQL aggregations for dashboards and alerting
 
-Currently, users must implement custom solutions outside of ADX-Mon to bridge this gap, leading to:
+Currently, users must either:
+- Use SummaryRules to materialize data in ADX tables, then build custom exporters to transform and export
+- Implement entirely separate infrastructure outside of ADX-Mon
+
+This leads to:
 - Duplicated infrastructure for metric transformation and export
 - Inconsistent metric schemas across different teams
-- Manual synchronization between SummaryRule execution and metrics export
+- Additional storage costs for intermediate table materialization
+- Complex multi-step pipelines that are difficult to maintain
 
 ## Solution Overview
 
-We propose implementing a feature that enables SummaryRule outputs to be automatically transformed and exported as OTLP metrics. This integration will provide a standardized, declarative way to create metrics pipelines from ADX data.
+We propose implementing a new `adxexporter` component that processes `MetricsExporter` CRDs to execute KQL queries and export results as metrics. This provides a streamlined, declarative way to create KQL-to-metrics pipelines with two complementary output modes:
+
+1. **Prometheus Scraping Mode** (Phase 1): Execute queries and expose results as Prometheus metrics on `/metrics` endpoint for Collector to scrape
+2. **Direct OTLP Push Mode** (Phase 2): Execute queries and push results directly to OTLP endpoints with backlog/retry capabilities
 
 ### Key Requirements
 
-1. **Generic Design**: Solution must work with any SummaryRule output, not tied to specific schemas
-2. **Standardized Transformation**: Define clear schema requirements for converting KQL results to OTLP metrics
-3. **Configurable Export**: Allow flexible configuration of export targets and cadence
-4. **Separation of Concerns**: Maintain clean boundaries between data aggregation and metrics export
+1. **Direct KQL Execution**: Execute KQL queries directly without requiring intermediate table storage
+2. **Dual Output Modes**: Support both Prometheus scraping and direct OTLP push patterns
```

Comment:
nit: Not sure that supporting both output modes is a key requirement, but do want at least one of em at a given time.